modules/core/src/main/scala/gem/Location.scala

```diff
@@ -0,0 +1,73 @@
+package gem
+
+import scalaz._, Scalaz._
+import scalaz.Ordering.{EQ, GT, LT}
+
+/** A sortable value used to indicate relative positions of a set of associated
+  * elements.  `Location`s may be thought of as lists of arbitrary integers
+  * which are be compared element by element such that the first pair of values
+  * that differ determine the ordering of the `Location`s as a whole.  If one
+  * `Location` is a proper prefex of another, then it sorts ahead of the other
+  * `Location`.
+  */
+sealed abstract case class Location(toNel: NonEmptyList[Int]) {
```

Comment:
This is specialized to `NonEmptyList[Int]` but you could make it work for any `F[A]` where some operations require `Foldable1[F]` and/or `Order[A]`. This would allow you to define `Foldable[Location]` which would give you `.toList` and `.toIList` for free. You may not want to go this far but it's an option.